account-kit/signer/src/base.ts

```diff
@@ -836,6 +864,53 @@ export abstract class BaseAlchemySigner<TClient extends BaseSignerClient>
     return new SolanaSigner(this.inner);
   };
 
+  private authenticateWithEmailNew = async (
+    params: Extract<AuthParams, { type: "email" }>
+  ): Promise<AuthStepResult> => {
+    if ("bundle" in params) {
+      const user = await this.completeEmailAuth(params);
+      return {
+        status: AlchemySignerStatus.CONNECTED,
+        user,
+      };
+    }
+
+    if (!("email" in params)) {
+      throw new Error("Email is required");
+    }
+
+    const { orgId, otpId, multiFactors, isNewUser } =
+      await this.initOrCreateEmailUser(
+        params.email,
+        params.emailMode ?? "otp",
+        params.multiFactors,
+        params.redirectParams
+      );
+
+    const isMfaRequired = multiFactors ? multiFactors?.length > 0 : false;
```

Comment:
(Very nit) No need for the `?.`, we know `multiFactors` is present in that branch.

Reply:
Oh yeah good catch